mimic/rest/maas_api.py

```diff
@@ -1264,3 +1347,144 @@ def multiplot(self, request, tenant_id):
         request.setResponseCode(200)
         self._audit('rollups', request, tenant_id, status, content)
         return json.dumps({'metrics': metrics_replydata})
+
+
+@implementer(IAPIMock, IPlugin)
+@attributes(["maas_api"])
+class MaasControlApi(object):
+    """
+    This class registers the MaaS controller API in the service catalog.
+    """
+    def catalog_entries(self, tenant_id):
+        """
+        List catalog entries for the MaaS API.
+        """
+        return [
+            Entry(
+                tenant_id, "rax: monitor", "cloudMonitoringControl",
+                [
+                    Endpoint(tenant_id, region, text_type(uuid4()),
+                             "v1.0")
+                    for region in self.maas_api._regions
+                ]
+            )
+        ]
+
+    def resource_for_region(self, region, uri_prefix, session_store):
+        """
+        Get an :obj:`twisted.web.iweb.IResource` for the given URI prefix;
+        implement :obj:`IAPIMock`.
+        """
+        maas_controller = MaasController(api_mock=self,
+                                         session_store=session_store,
+                                         region=region)
+        return maas_controller.app.resource()
+
+
+@attributes(["api_mock", "session_store", "region"])
+class MaasController(object):
+    """
+    Klein routes for MaaS control API.
+    """
+
+    def _entity_cache_for_tenant(self, tenant_id):
+        """
+        Retrieve the M_cache object containing all objects created so far
+        """
+        clock = self.session_store.clock
+        return (self.session_store.session_for_tenant_id(tenant_id)
+                .data_for_api(self.api_mock.maas_api, _mcache_factory(clock))[self.region])
+
+    app = MimicApp()
+
+    @app.route('/v1.0/<string:tenant_id>/entities/<string:entity_id>/alarms/test_response',
+               methods=['PUT'])
+    def set_test_alarm_response(self, request, tenant_id, entity_id):
+        """
+        Sets the test-alarm response for a given entity.
+        """
+        test_responses = self._entity_cache_for_tenant(tenant_id).test_alarm_responses
+        dummy_response = json.loads(request.content.read())
+        test_responses[entity_id] = []
+        for response_block in dummy_response:
+            ith_response = {'state': response_block['state']}
+            if 'status' in response_block:
+                ith_response['status'] = response_block['status']
+            test_responses[entity_id].append(ith_response)
+        request.setResponseCode(204)
+        return ''
```

Comment:
I'd like to address this separately as the idiom is peppered throughout the MaaS API.